### PR TITLE
Record how long it takes BatchManager to complete full batch

### LINKED ISSUE #8214: Log how long it takes ScheduleManager to complete batch / pause op processing
We should measure and report (when it takes too long) how long it takes for ScheduleManager to wait (and keep ops in paused state) before it can complete a batch. We can leverage same technique as with op latency events to log error if it's longer than 5s.

### PR DESCRIPTION
**Pull request by vladsud**
Make processing of ops async, the same way it is in DeltaQueue.
This helps with UTs actually testing what we ship, but also (due to to small change in order in BatchManager) is now asserted by BatchManager, i.e. no reentrancy is allowed by inbound queue being synchronous.
Some renames in UT due to this separation.

And the main change - add error if it takes us more than 5 seconds to get full batch. The number is arbitrary, and I we will likely change it in future (as we will hit these errors in normal flow, which means they should not be errors). But for now, this is in sync with "OpRoundtripTime" errors that report latency delays as errors at 5 second mark, and the primary reason for this change.

Will close #8214 
Helps with continued investigation of #7406
Natural continuation of PR #8206

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -386,24 +389,37 @@ class ScheduleManagerCore {
...
             // If the next op is the start of incomplete batch, then we can't process it until it's fully in - pause!
             if (sequenceNumber + 1 === this.pauseSequenceNumber) {
-                this.setPaused(true);
+                this.setPaused();
             }
         }
     }
 
-    private setPaused(localPaused: boolean) {
+    private setPaused() {
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:397]: Can we call these `pauseQueue` and `resumeQueue` or something like that. setPaused / setResumed are weird :).

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -386,24 +389,37 @@ class ScheduleManagerCore {
...
+        assert(!this.localPaused, "always called from resumed state");
+        this.localPaused = true;
+        this.timePaused = performance.now();
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        this.deltaManager.inbound.pause();
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:402]: Is it okay to not await this? It seems risky that pause() is async. What if the queue is paused later and some ops pass through?

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -386,24 +389,37 @@ class ScheduleManagerCore {
...
-        this.localPaused = localPaused;
-        if (localPaused) {
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            this.deltaManager.inbound.pause();
-        } else {
-            this.deltaManager.inbound.resume();
+        this.localPaused = false;
+        const duration = performance.now() - this.timePaused;
+        // Random round number - we want to know when batch waiting paused op processing.
+        if (duration > 5000) {
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:414]: nit: Consider declaring a global variable with this value `5000`.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 4]
```diff
@@ -386,24 +389,37 @@ class ScheduleManagerCore {
...
+            this.logger.sendErrorEvent({
+                eventName: "BatchWaitTime",
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:416]: nit: `MaxBatchWaitTimeExceeded`

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 5]
```diff
@@ -94,63 +94,81 @@ describe("Runtime", () => {
                     sequenceNumber = 0;
                 });
 
-                function processOp(partialMessage: Partial<ISequencedDocumentMessage>) {
+                /**
+                 * Pushes single op to the inbound queue. Adds proper sequence numbers to them
+                 */
+                function inboundOp(partialMessage: Partial<ISequencedDocumentMessage>) {
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:100]: nit: Call this `pushOp`

**Review by agarwal-navin — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 2 above)
Comment (vladsud) [reply packages/runtime/container-runtime/src/containerRuntime.ts:402]: That should not be possible in UT, as all of the test cases assume that UT is the only entity controlling pausing, and we know we use mock implementation that pauses it right away. But I'll add awaits - it does not change anything

**Pull request merged**